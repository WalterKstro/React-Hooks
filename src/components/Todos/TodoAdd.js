import PropTypes from 'prop-types';
import useForm from "../../hooks/useForm"

const TodoAdd = ({handleAdd}) => {

    const {state, handleChangeInput, handleReset} = useForm({task: ''})
    const {task} =  state

    const handleSubmit = e => {
        e.preventDefault()
        if(task !== '') {
            const todo = {id: new Date().getTime(), task, state : false}
            handleAdd(todo)
            handleReset()
        }
    } 

    return (
        <form onSubmit={handleSubmit}>
             <input 
                type="text" 
                className="form-control mb-3" 
                placeholder="Quiero aprender ..." 
                name="task"
                value={task}
                onChange={handleChangeInput}/>
         </form>
    )
}

TodoAdd.propTypes = {
    handleAdd: PropTypes.func.isRequired
}

export default TodoAdd
