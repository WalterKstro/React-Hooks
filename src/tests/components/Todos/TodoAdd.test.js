import { shallow } from "enzyme"
import TodoAdd from "../../../components/Todos/TodoAdd"

describe('Testing in componente <TodoAdd/>', () => {

    let component
    let handleAdd
    beforeEach(() => {
        handleAdd = jest.fn()
        component = shallow(<TodoAdd handleAdd={handleAdd}/>)
    })

    test('should show correctly the component', () => {
        expect(component).toMatchSnapshot()
    })
    
    test('Not should call the function handleAdd', () => {
        /** Una opciÃ³n similar a simulate() */
        const handleSubmit = component.find('form').prop('onSubmit')
        handleSubmit({ preventDefault(){} })

        expect(handleAdd).toBeCalledTimes(0)
    })


    test('should call the function handleAdd', () => {
        const value = 'Learning Javascript'
        
        component.find('input').simulate('change', {
            target: {
                name: 'task',
                value
            }
        })

        const handleSubmit = component.find('form').prop('onSubmit')
        handleSubmit({ preventDefault(){} })

        expect(handleAdd).toBeCalledTimes(1)
        expect(handleAdd).toHaveBeenCalledWith({
            task: value,
            state: false,
            id: expect.any(Number)
        })

        expect(component.find('input').prop('value')).toBe('')

        
    })
    
    
    
})


