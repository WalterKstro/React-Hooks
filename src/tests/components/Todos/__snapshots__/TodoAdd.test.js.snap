// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in componente <TodoAdd/> should show correctly the component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="form-control mb-3"
    name="task"
    onChange={[Function]}
    placeholder="Quiero aprender ..."
    type="text"
    value=""
  />
</form>
`;
